### PR TITLE
Improved jit client

### PR DESCRIPTION
**Pull request by NourAlharithi**

### CODE REVIEW

File: programs/jit-proxy/src/lib.rs [hunk 1]
```diff
@@ -70,13 +70,21 @@ pub mod jit_proxy {
...
                 }
             }
             PositionDirection::Short => {
                 if taker_price < maker_worst_price {
-                    msg!("taker price {} < worst ask {}", taker_price, maker_worst_price);
+                    msg!(
+                        "taker price {} < worst ask {}",
+                        taker_price,
+                        maker_worst_price
+                    );
```
Comment (0xbigz) [inline programs/jit-proxy/src/lib.rs:87]: can you revert this? just dont want to mix cross-code formatting with PR as much as possible

File: ts/sdk/src/jitter.ts [hunk 2]
```diff
@@ -136,57 +188,256 @@ export class Jitter {
...
+		initialDetails: AuctionAndOrderDetails
+	): Promise<{ slot: number; updatedDetails: AuctionAndOrderDetails }> {
+		const auctionEndSlot = order.auctionDuration + order.slot.toNumber();
+		let currentDetails: AuctionAndOrderDetails = initialDetails;
+		let willCross = initialDetails.willCross;
+		return new Promise((resolve) => {
+			// Immediately return if we are past target slot
+
+			// Otherwise listen for new slots in case we hit the target slot and we're gonna cross
+			this.slotSubscriber.eventEmitter.on('newSlot', (slot) => {
```
Comment (crispheaney) [inline ts/sdk/src/jitter.ts:420]: do you need to delete these callbacks after the auction expires? over time it'll get slower and slower if there are a bunch of callbacks?

Comment (crispheaney) [reply ts/sdk/src/jitter.ts:420]: or mybe it should only be on callback?

Comment (NourAlharithi) [reply ts/sdk/src/jitter.ts]: fixed this

File: programs/jit-proxy/src/lib.rs (hunk 1 above)
Comment (NourAlharithi) [reply programs/jit-proxy/src/lib.rs:87]: unfortunately not bc of the pre commit hooks which i'm trying not to change

File: ts/sdk/src/jitter/jitterSniper.ts [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+	async waitForSlotOrCrossOrExpiry(
+		targetSlot: number,
+		order: Order,
+		initialDetails: AuctionAndOrderDetails
+	): Promise<{ slot: number; updatedDetails: AuctionAndOrderDetails }> {
+		const auctionEndSlot = order.auctionDuration + order.slot.toNumber();
+		let currentDetails: AuctionAndOrderDetails = initialDetails;
+		let willCross = initialDetails.willCross;
+		return new Promise((resolve) => {
+			// Immediately return if we are past target slot
```
[the remark this answers is not in the document]
Comment (NourAlharithi) [reply ts/sdk/src/jitter/jitterSniper.ts:296]: done

**Pull request merged**